Intro to R Revamp (PR 1 of n) : 00-setup-table.R script

cansavvy opened the pull request:
## Background
We are revamping the Intro to R materials. See issue #162 
First part of this involves setting up a nice and clean table of real data for participants to use. 

## Summary of the methods in the script 
I added a microarray dataset of pilocytic astrocytoma tumour samples with matched cerebellum controls. 
1) I read in all the SCAN normalized array data that I downloaded from refinebio. 
2) I used limma on default settings to do a model between the cerebellum and tumors 
3) I clean up the table and add a gene_symbols column 
4) The resulting table is saved to a `data/gene_results_GSE44971.tsv` folder and has the following columns: 
```
"ensembl_ids"     "gene_symbols"    "avg_expression"  "log_fold_change" "t_value"         "p_value" "adj_p_value"     "log_odds" 
```

In the upcoming PR I will build a notebook around plotting and manipulating the data in this table. 

## About the current dataset
Note I am not totally sold on this dataset in particular, but we can pretty easily switch it out with another dataset if we end up not liking it. 
Right now I'm more focused on the methods setting up.
I retrieved this dataset from [refine.bio](https://www.refine.bio/experiments/GSE44971/gene-expression-data-from-pilocytic-astrocytoma-tumour-samples-and-normal-cerebellum-controls)
In forthcoming PRs, I will add a paragraph describing these data briefly so the participants are aware of what these results are based on. 
Here is the paper: 
[lambert2013.pdf](https://github.com/AlexsLemonade/training-modules/files/4355310/lambert2013.pdf)

## Questions for reviewers: 
- How do we feel about the resulting table? Should any columns be added or taken away? 
- What do you think about the scripts/ folder add and folder organization in general? My plan, based on @jaclyn-taroni 's idea is to have links to the script in the participant facing notebook so if participants want to know how this table was made they will have this available to them, but it won't require them to have to parse through these steps right off the bat. 
- Is the model set up I selected here okay and easy enough to follow? 
- Should I delete the old data files and notebooks now? I haven't as of yet. Figured I will replace each notebook as I feel its material has been superceded by the revamped notebooks.

jashapiro commented on intro-to-R-tidyverse/scripts/00-setup-intro-to-R.R:
```diff
@@ -0,0 +1,88 @@
+  # Move ensembl IDs to their own column
+  tibble::rownames_to_column("ensembl_ids") %>%
+  # Map ensembl IDs to their associated gene symbols
+  dplyr::mutate("gene_symbols" = AnnotationDbi::mapIds(
```
```suggestion
  dplyr::mutate("gene_symbol" = AnnotationDbi::mapIds(
```

jashapiro commented on intro-to-R-tidyverse/scripts/00-setup-intro-to-R.R:
```diff
@@ -0,0 +1,88 @@
+# Apply multiple testing correction and obtain stats
+stats_df <- limma::topTable(fit, number = nrow(gene_df)) %>% 
+  # Move ensembl IDs to their own column
+  tibble::rownames_to_column("ensembl_ids") %>%
```
```suggestion
  tibble::rownames_to_column("ensembl_id") %>%
```

jashapiro commented on intro-to-R-tidyverse/scripts/00-setup-intro-to-R.R:
```diff
@@ -0,0 +1,88 @@
+  dplyr::select(metadata$geo_accession)
+
+# Create the design matrix from the genotype information
+des_mat <- model.matrix(~metadata$tissue)
```
What happens if we make this a bit more complex? I can't remember what limma output will look like, but why not tissue* sex for a simple factorial design? Sample size in this data set may get a bit small, but from a quick glance, it appears there are 3 in the smallest group (Female normal). And we all know 3 is the correct number.

If this works out, you may have to add a bit more cleaning to *keep it tidy™*.

cansavvy replied:
Okay. I'm in a bit of a `model.matrix` and `limma` rabbit hole. What is our main question(s) and contrasts we want to make? Is this correct? 
1) Differences between normal and tumor
2) Differences between sexes
3) Interaction of sex and tumor?

cansavvy replied:
After playing around with this, I feel like doing a 2x2 design makes this more complicated than needed when this module is mainly focused on getting the participants to feel comfortable with data.frames and ggplot2. I don't necessarily want to add an added layer of complexity that seems to not add toward the main learning objective. What do you think @jashapiro ?

jashapiro replied:
That would be fine. Might as well do it correctly, which would have all three of those.
So you ought to be able to just do: 
```
des_mat <- model.matrix(~metadata$tissue * metadata$refinebio_sex)
```
to get all of those in the model.

cansavvy replied:
^I mean yes, however if you want the logFC for each comparison, `toptable` will not automatically do that. And then I went down a rabbit hole of what comparisons do we care about.

cansavvy replied:
I just feel like its very easy to get distracted by this 2x2 experimental model and I am not sure (at this point in time) how this aids the participant's learning of tidyverse. Maybe I need a clearer idea of your vision behind doing a 2x2 model.

jashapiro replied:
A fair point. My vision is actually pretty simple here. I just want more than a single comparison, so we can filter results down to just one comparison for plotting at first, then show them how we can make multiples with facet_grid. 

The idea is to give little hints at the power available. Not go down rabbit holes.

cansavvy replied:
Alright. I tried to follow 8.7 instead. The contrasts probably need an extra pair of eyes to check.

cansavvy commented on intro-to-R-tidyverse/scripts/00-setup-intro-to-R.R:
```diff
@@ -0,0 +1,127 @@
+  dplyr::select(metadata$geo_accession)
+
+#This set up is based on the example 11.4 Estrogen Data: A 2x2 Factorial Experiment in 
+# https://bioc.ism.ac.jp/packages/2.9/bioc/vignettes/limma/inst/doc/usersguide.pdf
```
I was following this set up, @jashapiro whether or not this is correct for the questions we are asking, I do not know.

jashapiro replied:
That is okay, but you'll note they are only interested in treatment effects at each time point, not differences between time points. A better option for this would be the setup in section 8.7 of that doc, probably mostly the one in the bottom of page 49. This would give the mean sex effect, mean tumor-normal effect, and the interaction effect.

cansavvy replied:
Okay. Great. I will model it after that one then. Thanks!

cansavvy commented:
@jashapiro , Ready for another look!

cansavvy commented on intro-to-R-tidyverse/scripts/00-setup-intro-to-R.R:
```diff
@@ -48,46 +48,36 @@ metadata <- readr::read_tsv(file.path(data_dir,
+fit2 <- limma::eBayes(fit)
+
+# collect results for each contrast
+stats_df <- dplyr::bind_rows(
```
If we are going to have participants looking at this script at all, we might want to separate out these steps a bit so they are easier to follow. I'll go ahead and separate it out if that is fine with you @jashapiro

cansavvy replied:
Not that we are planning on lecturing on this at all, but just that it will be included for their reference, so don't want them to be overwhelmed if they open it.

cansavvy commented:
@jashapiro , made some minor changes in response to your changes. Ready for another glance. Here's what I did: 
- Changed column name from `t` -> `t_statistic` for disambiguation purposes
- Added a lot of comments so if participants look at this they'll have more info on each step
- I stretched out your bind_rows step to a few steps.

jashapiro commented on intro-to-R-tidyverse/scripts/00-setup-intro-to-R.R:
```diff
@@ -0,0 +1,137 @@
+    gene_symbol,
+    contrast,
+    avg_expression = AveExpr, # We want our column names to be consistent format
+    t_statistic = t, # There is a function called `t` so for disambiguation purposes, we will name this t_value
```
```suggestion
    t_statistic = t, # There is a function called `t` so for disambiguation purposes, we will name this t_statistic
```
